core/src/apps/evolu/evolu_sign_registration_request.py

```diff
@@ -55,35 +49,47 @@ async def evolu_sign_registration_request(
     else:
         raise RuntimeError("Optiga is not available")
 
+    challenge_bytes, size_bytes = _check_data(
+        msg.challenge_from_server, msg.size_to_acquire
+    )
+
     if not check_delegated_identity_proof(
         provided_proof=bytes(msg.proof_of_delegated_identity),
         header=b"EvoluSignRegistrationRequest",
         arguments=[
-            bytes(msg.challenge_from_server),
-            msg.size_to_acquire.to_bytes(4, "big"),
+            challenge_bytes,
+            size_bytes,
         ],
     ):
         raise ValueError("Invalid proof")
 
-    private_key = get_delegated_identity_key()
-    public_key = get_public_key_from_private_key(private_key)
+    signature = _get_signature(challenge_bytes, size_bytes)
+    certificates = _get_certificates()
 
-    header = b"EvoluSignRegistrationRequestV1:"
-    components = [
-        header,
-        public_key,
-        msg.challenge_from_server,
-        msg.size_to_acquire.to_bytes(4, "big"),
-    ]
-    h = utils.HashWriter(sha256())
-    for component in components:
-        write_compact_size(h, len(component))
-        h.extend(component)
+    return EvoluRegistrationRequest(
+        certificate_chain=certificates,
+        signature=signature,
+    )
+
+
+def _check_data(challenge: bytes, size: int) -> tuple[bytes, bytes]:
+    from trezor import wire
+
+    if not (1 <= len(challenge) <= 255):
+        raise wire.DataError("Invalid challenge length")
+    if size < 0 or size > 256**4 - 1:
```

Comment:
please, use a constant instead of formula

```suggestion
    if size < 0 or size > 0xFFFFFFFF:
```